### PR TITLE
UCP/EP: Don't call error callback for internal endpoints

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7023/ for details (Mellanox internal link).

File: src/ucp/core/ucp_worker.c [hunk 1]
```diff
@@ -417,8 +417,13 @@ static unsigned ucp_worker_iface_err_handle_progress(void *arg)
     ucp_ep->am_lane   = 0;
 
     if (ucp_ep_ext_gen(ucp_ep)->err_cb != NULL) {
+        ucs_debug("ep %p: calling user error callback %p with arg %p", ucp_ep,
```
Comment (evgeny-leksikov) [inline src/ucp/core/ucp_worker.c]: minor: pls add `ucs_assert(ucp_ep->flags & UCP_EP_FLAG_USED)` into branch above to clarify that `err_cb` can be set only by API call

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7029/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7044/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7052/ for details (Mellanox internal link).

**Pull request merged**